documentcloud.php

```diff
@@ -25,73 +25,104 @@
+		if ( ! defined( 'WPCOM_IS_VIP_ENV' ) || ! WPCOM_IS_VIP_ENV ) {
+			add_action( 'admin_init', array( $this, 'check_dc_plugin_conflict') );
+	        add_action( 'init', array( $this, 'register_dc_oembed_provider' ) );
+	    }
 
-        add_action('admin_init', array(&$this, 'check_dc_plugin_conflict'));
+	    // Set the textdomain for the plugin so it is translation compatible
+		add_action( 'plugins_loaded', array( $this, 'load_plugin_textdomain' ) );
 
-        add_action('init', array(&$this, 'register_dc_oembed_provider'));
         // Only called when `[documentcloud]` shortcode is used
-        add_shortcode('documentcloud', array(&$this, 'process_dc_shortcode'));
+        add_shortcode( 'documentcloud', array( $this, 'process_dc_shortcode' ) );
+
         // Called just before oEmbed endpoint is hit
-        add_filter('oembed_fetch_url', array(&$this, 'prepare_oembed_fetch'), 10, 3);
+        add_filter( 'oembed_fetch_url', array( $this, 'prepare_oembed_fetch' ), 10, 3 );
 
-        // Setup admin settings
-        add_action('admin_menu', array(&$this, 'add_options_page'));
-        add_action('admin_init', array(&$this, 'settings_init'));
+        // Setup the settings page
+        add_action( 'admin_menu', array( $this, 'add_options_page' ) );
+        add_action( 'admin_init', array( $this, 'settings_init' ) );
 
-        // Store metadata upon post save
-        add_action('save_post', array(&$this, 'save'));
+        // Store Documentcloud metadata upon post save
+        add_action( 'save_post', array( $this, 'save' ), 10, 2 );
     }
-    
+
+	/**
+	 * Load plugin textdomain.
+	 */
+	function load_plugin_textdomain() {
+		load_plugin_textdomain( 'documentcloud' );
+	}
+
+	/**
+	 * Check for conflicts with the Navis Documentcloud plugin.
+	 */
     function check_dc_plugin_conflict() {
-        if (is_plugin_active('navis-documentcloud/navis-documentcloud.php')) {
-            add_action( 'admin_notices', array(&$this, 'dc_conflict_admin_notice'));
+        if ( is_plugin_active( 'navis-documentcloud/navis-documentcloud.php' ) ) {
+            add_action( 'admin_notices', array( $this, 'dc_conflict_admin_notice' ) );
         }
     }
 
+	/**
+	 * Create an admin notice when conflicts exist with Navis Documentcloud.
+	 */
     function dc_conflict_admin_notice() {
         ?>
         <div class="error">
-            <p><?php _e( '<b>Warning!</b> You have two conflicting DocumentCloud plugins activated. Please deactivate Navis DocumentCloud, which has been replaced by <a target="_blank" href="https://wordpress.org/plugins/documentcloud/">DocumentCloud</a>.', 'documentcloud-plugin-conflict' ); ?></p>
+            <p><?php echo wp_kses( __( '<b>Warning!</b> You have two conflicting DocumentCloud plugins activated. Please deactivate Navis DocumentCloud, which has been replaced by <a target="_blank" href="https://wordpress.org/plugins/documentcloud/">DocumentCloud</a>.', 'documentcloud' ) ); ?></p>
```

Comment:
@bcampeau I wasn't familiar with `wp_kses()`, but my reading of its [docs](https://codex.wordpress.org/Function_Reference/wp_kses) indicate we need to pass in a whitelist of HTML tags. Does it work fine without?